mobile-widgets/qmlmanager.cpp

```diff
@@ -170,11 +171,12 @@ QMLManager::QMLManager() : m_locationServiceEnabled(false),
 			this, &QMLManager::uploadProgress);
 
 	// uploadDiveShare::uploadFinish() is defined with 3 parameters,
-	// whereas QMLManager::uploadFinish() is defined with 2 paramters,
+	// whereas QMLManager::uploadFinish() is defined with 2 parameters,
 	// Solution add a slot as landing zone.
-	connect(uploadDiveShare::instance(), SIGNAL(uploadDiveShare::uploadFinish(bool, const QString &, const QByteArray &)),
-			this, SLOT(uploadFinishSlot(bool, const QString &, const QByteArray &)));
+	connect(uploadDiveShare::instance(), &uploadDiveShare::uploadFinish,
+			this, &QMLManager::uploadFinishSlot);
 
+	void uploadFinish(bool success, const QString &text, const QByteArray &html);
```

Comment:
I'm surprised by this declaration in the middle of the C++ file. Can you explain the logic here?

Reply:
me too, I wonder why the compiler did not catch this.

It is a leftover copy/paste.

Removed,

Thanks for catching this.